components/Comment/CommentEditor.tsx

```diff
@@ -53,17 +59,40 @@ const CommentEditor = ({
     content,
   });
 
+  if (previewWhenInactive) {
+    useEffectHandleClick({
+      el: editorRef.current,
+      onOutsideClick: () => {
+        setIsPreviewMode(true);
+        isPreviewModeRef.current = true;
+      },
+      onInsideClick: () => {
+        setIsPreviewMode(false);
+        isPreviewModeRef.current = false;
+        quill && !quill.hasFocus() && quill.focus();
+      },
+    });
+  }
+
   useEffect(() => {
-    const isDisabled = isQuillEmpty(_content) ? true : false;
-    setIsSubmitDisabled(isDisabled);
+    const isEmpty = isQuillEmpty(_content) ? true : false;
+    setIsEmpty(isEmpty);
+    isEmptyRef.current = isEmpty;
```

Comment:
Is this ref also being used? Not seeing it being invoked anywhere

Reply:
It is used. Let me huddle you and walk through this.